packages/core/src/zone/ng_zone.ts

```diff
@@ -142,7 +146,10 @@ export class NgZone {
       self._inner = self._inner.fork((Zone as any)['longStackTraceZoneSpec']);
     }
 
-    self.shouldCoalesceEventChangeDetection = shouldCoalesceEventChangeDetection;
+    if (!shouldCoalesceChangeDetection) {
+      self.shouldCoalesceEventChangeDetection = shouldCoalesceEventChangeDetection;
+    }
```

Comment:
The above code makes the `shouldCoalesceEventChangeDetection` property conditional. (as in sometimes it is not present) This will make the code megamorphic as the shape of the `NgZone` will be different depending on how it is called. It is best practice to make sure that all properties are always defined and set it to `null`/`undefined` if not needed.

```suggestion
    self.shouldCoalesceEventChangeDetection = !shouldCoalesceChangeDetection 
 ? shouldCoalesceEventChangeDetection : false;
```

Reply:
Got it, thanks.